### ISSUE TITLE
Linux add agent -FAIL- no https

### ISSUE THREAD

**Issue by theredkoala21** [Ylianst/MeshCentral#441]
When adding new agents via the included 'Add agent script'. It display's an error and is unable to continue.

**From first glance, could the fact that there isn't a --no-check-certificate in the wget options // option -k for curl?**

It display's the error:

```
root@piw2:/# (wget https://mesh.notadomain.com:8443/meshagents?script=1 --no-check-certificate -O ./meshinstall.sh || wget https://mesh.notadomain.com:8443/meshagents?script=1 --no-proxy --no-check-certificate -O ./meshinstall.sh) && chmod 755 ./meshinstall.sh && sudo ./meshinstall.sh https://mesh.notadomain.com:8443 'somesecurestringthatisasecretorsomething'
--2019-08-21 08:09:31--  https://mesh.notadomain.com:8443/meshagents?script=1
Resolving mesh.notadomain.com (mesh.notadomain.com)... 44.55.66.77
Connecting to mesh.notadomain.com (mesh.notadomain.com)|44.55.66.77|:8443... connected.
WARNING: The certificate of ‘mesh.notadomain.com’ is not trusted.
WARNING: The certificate of ‘mesh.notadomain.com’ doesn't have a known issuer.
HTTP request sent, awaiting response... 200 OK
Length: 9194 (9.0K) [text/plain]
Saving to: ‘./meshinstall.sh’

./meshinstall.sh              100%[==============================================>]   8.98K  --.-KB/s    in 0.001s

2019-08-21 08:09:31 (6.95 MB/s) - ‘./meshinstall.sh’ saved [9194/9194]

Failed to disable unit: Unit file meshagent.service does not exist.
Failed to stop meshagent.service: Unit meshagent.service not loaded.
Agent uninstalled.
Downloading Mesh agent #25...
--2019-08-21 08:09:31--  https://mesh.notadomain.com:8443/meshagents?id=25
Resolving mesh.notadomain.com (mesh.notadomain.com)... 44.55.66.77
Connecting to mesh.notadomain.com (mesh.notadomain.com)|44.55.66.77|:8443... connected.
ERROR: The certificate of ‘mesh.notadomain.com’ is not trusted.
ERROR: The certificate of ‘mesh.notadomain.com’ doesn't have a known issuer.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

```

It seems the error is displayed because of the ssl certificate error from curl. 

Im assuming this would happen for most linux systems, but for this example i was using Raspbian Buster / 2019-07-10.

**Comment by theredkoala21** [comment]
I think i found the issue:
```
    function isTrustedCert() {
        if (obj.args.notls == true) return false; // We are not using TLS, so not trusted cert.
        if (obj.args.tlsoffload != null) return true; // We are using TLS offload, a real cert is likely used.
        if (obj.parent.config.letsencrypt != null) return true; // We are using Let's Encrypt, real cert in use.
        if (obj.certificates.WebIssuer.indexOf('MeshCentralRoot-') == 0) return false; // Our cert is issued by self-signed cert.
        if (obj.certificates.CommonName.indexOf('.') == -1) return false; // Our cert is named with a fake name
        return true; // This is a guess
    }
```
[webserver.js](https://github.com/Ylianst/MeshCentral/blob/86d19fadbfe44e026e9fc51bbc335adaf8ed4e62/webserver.js)


The certificate i use is a self signed ca approved internal certificate. It passes the test for a 'TrustedCert'. Can we implement a config.json override, cause it is more pain than its worth to worry about the weird cases like a self signed non authed-ca approved certificate lol.

**Comment by Ylianst** [comment]
Published MeshCentral v0.4.0-d. Starting with this version, you can just tell the server if the TLS certificate is good or not. Use the "trustedcert" option in the "settings" section, or in one of the "domain" sub-sections. Let me know if that fixes the problem.

```
{
  "settings": {
    "Cert": "myserver.mydomain.com",
    "TrustedCert": false
  }
}
```

**Comment by theredkoala21** [comment]
Yes. Excellent. Thanks
